homeassistant/components/plant.py

```diff
@@ -250,6 +247,14 @@ def _check_max(self, sensor_name, value, params):
             # only use the database if it's configured
             self.hass.async_add_job(self._load_history_from_db)
 
+        async_track_state_change(self.hass, list(self._sensormap),
+                                 self.state_changed)
+
+        for entity_id in self._sensormap:
+            state = self.hass.states.get(entity_id)
+            if state is not None:
+                self.state_changed(entity_id, None, state)
```

Comment:
Maybe we want to wait with this to home assistant has started? Then we can be sure that all the tracked entities are set up. We could break out this block into a function which we add as a callback to home assistant start event listener. The event listerer should be subscribed from `async_added_to_hass`.

Reply:
Interesting ... my thinking was **not** to ensure that all of the tracked entities had been set-up but rather to deal with any state changes that might have occurred before the plant entity had a chance to start listening. Any subsequently created tracked entities will result in a listener callback.

Do you see scenarios where this wouldn't be the case ?

Reply:
When an entity is added there should always be a new state callback trigger, so that should work. If the state is unknown at that time, we won't get a reading, but we can't solve this with waiting until home assistant start, since the clock tick won't start until that time anyway, so polling updates won't start earlier.

So your point is good!